src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -282,10 +283,10 @@ public static boolean isItemSimilar(@Nullable ItemStack item, @Nullable ItemStac
             ItemMeta itemMeta = item.getItemMeta();
 
             if (sfitem instanceof SlimefunItemStack) {
-                Optional<String> id = Slimefun.getItemDataService().getItemData(itemMeta);
+                String id = getMetaId(itemMeta);
 
-                if (id.isPresent()) {
-                    return id.get().equals(((SlimefunItemStack) sfitem).getItemId());
+                if (id != null) {
+                    return id.equals(((SlimefunItemStack) sfitem).getItemId());
```

Comment:
We probably want a distinctive check here too. Since this is also just an ID check currently

not sure when this code is called but hey, will probably solve issues in addons or other parts of the code

Reply:
Added